.github/workflows/wasmcloud.yml

```diff
@@ -81,6 +81,40 @@ jobs:
         name: wasmcloud-universal-darwin
         path: wasmcloud-universal-darwin
 
+  oci:
+    runs-on: ubuntu-22.04
+    permissions:
+      packages: write
+    needs: build-bin
+    steps:
+    - uses: actions/checkout@v3
+    - uses: ./.github/actions/install-nix
+      with: 
+        cachixAuthToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
+    - uses: docker/login-action@v2
+      if: ${{ github.event_name != 'pull_request' }}
+      with:
+        registry: ghcr.io
+        username: ${{ github.repository_owner }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+    - run: nix run -L .#build-wasmcloud-oci wasmcloud
+    - run: nix profile install --inputs-from . 'nixpkgs#buildah'
+      if: ${{ github.event_name != 'pull_request' }}
+    - run: buildah push --storage-driver=vfs --all --format 'v2s2' wasmcloud docker://ghcr.io/${{ github.repository_owner }}/wasmcloud:${{ github.sha }}
```

Comment:
I'm not familiar with this. Is there a reason we can't push to wasmcloud.azurecr.io, where our release images are?

Reply:
We have already agreed to push to GitHub registry and Docker Hub.
Docker Hub will be the next step.
Please file an issue for pushing to azure

Reply:
Can you help me understand the reasoning for this/point me to where we agreed to push to GHCR? I'm unfamiliar with `buildah` and whether changing to ACR is more than swapping the string here

Reply:
We have discussed that at standup.
No, this is not much different from `docker push`

Reply:
From what I remember from the discussion, we decided on GHCR so that the image could be discoverable from the packages tab on GitHub and that all of the release assets could be centralized. Also, Dockerhub is good because when people think of a container image they go to dockerhub. I think we put AzureCR as a third priority because the content discovery isn't great there, so people would only know what's there based on what we explicitly document.

Good to know buildah is similar to a docker push so we can just add those here as we put the correct credentials into secrets.

Reply:
Yes, in fact, that's already done in the PR, we push to all three of them, e.g. https://hub.docker.com/repository/docker/rvolosatovs/wasmcloud/general ( don't have an Azure account)